File: tests/cpp/binary_search_trees/test_same_bsts.cpp
#include "../../../problems/binary_search_trees/same_bsts/cpp/same_bsts.h"
#include "../jsontestbase.h"

#include <vector>

class TestSameBSTs : public JsonTestBase {
public:
  TestSameBSTs() {
    json_file_path =
        std::string(TEST_CASES_DIR) + "/binary_search_trees/same_bsts.json";
  }
};

TEST_F(TestSameBSTs, TestCases) {
  ASSERT_FALSE(test_cases.empty()) << "Test cases are empty";

  for (const auto &test : test_cases) {
    auto arrayOne = test["arrayOne"].get<std::vector<int>>();
    auto arrayTwo = test["arrayTwo"].get<std::vector<int>>();
    auto expected = test["expected"].get<bool>();

    auto result = sameBsts(arrayOne, arrayTwo);

    EXPECT_EQ(result, expected) << "Failed for test: " << test.dump(2);
  }
}

File: tests/python/binary_search_trees/same_bsts/test_same_bsts.py
from problems.binary_search_trees.same_bsts.python.same_bsts import sameBsts
from tests.python.utility import load_test_cases


def test_same_bsts() -> None:
    test_cases = load_test_cases("binary_search_trees/same_bsts.json")

    for idx, test in enumerate(test_cases):
        arrayOne = test["arrayOne"]
        arrayTwo = test["arrayTwo"]
        expected = test["expected"]

        result = sameBsts(arrayOne, arrayTwo)

        assert (
            result == expected
        ), f"Test case {idx} failed: expected {expected}, got {result}"
